sections/security/userpasswords.md

```diff
@@ -0,0 +1,115 @@
+# Secure Your Users' Passwords
+
+### One Paragraph Explainer
+
+**Always** encrypt users' passwords as opposed to storing them as text; there are three options that depend on your use case for encrypting user passwords. All the below functions need to be implemented properly to provide any security. (Reference the miminums or see the [IETF's recommendations](https://tools.ietf.org/id/draft-whited-kitten-password-storage-00.html#name-kdf-recommendations) for what parameters to use for each) You should use the recommended properties as a minimum, using higher parameters and a combination that's unique to your program can help mitigate some of the damage if someone ever runs off with your password hashes. Also, always add a [salt](https://auth0.com/blog/adding-salt-to-hashing-a-better-way-to-store-passwords/) (reproducable data, unique to the user and your system) to your passwords before you hash.

```

Comment:
I think you meant to say hashing here?
Generally speaking, all 3 options: encrypting, hashing, and plain-text are valid to say but encryption !== hashing.

Reply:
Lol, yep. I even added a link to the differences between hashing and encryption in the references, but then went said it wrong right at the top.

Reply:
No worries, all good :-)